packages/react/src/AdvancedVideo.tsx

```diff
@@ -115,7 +116,12 @@ class AdvancedVideo extends Component <VideoProps> {
       if (key in this.props) {
         result[key] = this.props[key];
       }
-    });
+    })
+    if (this.props.cldPoster === 'auto') {
+      result['poster'] = this.props.cldVid.setAssetType('video').delivery("q_auto").format("auto:image").toURL();
```

Comment:
won't this alter `this.props.cldVid` ? maybe we want to duplicate it? keep it intact?

Reply:
Turns out that this `setAssetType` can be removed and it works the same!
I used a code example from the task description but it might have been inaccurate.